src/gfx/renderer_types.hpp

```diff
@@ -1,12 +1,38 @@
-#ifndef GFX_RENDERER_TYPES
-#define GFX_RENDERER_TYPES
+#ifndef CA95E36A_DF18_4EE1_B394_4094F976B20E

```

Comment:
tl;dr Moved internal structs used within renderer.cpp to this header.
Also added some new structs namely:
- PipelineDrawables
- RenderTargetData
- RenderGraphNode/RenderGraph/RenderGraphEvaluator